Error building DKMS module for kernel 6.4.0-rc1 kvmfr/0.0.8

On Arch Linux when upgrading the [mainline kernel](https://aur.archlinux.org/packages/linux-mainline) I saw the following error.  

```
Install DKMS modules
==> dkms install --no-depmod kvmfr/0.0.8 -k 6.4.0-rc1-1-mainline
Error! Bad return status for module build on kernel: 6.4.0-rc1-1-mainline (x86_64)
Consult /var/lib/dkms/kvmfr/0.0.8/build/make.log for more information.
==> WARNING: `dkms install --no-depmod kvmfr/0.0.8 -k 6.4.0-rc1-1-mainline' exited 10
```

`/var/lib/dkms/kvmfr/0.0.8/build/make.log`

```log
   1   │ DKMS make.log for kvmfr-0.0.8 for kernel 6.4.0-rc1-1-mainline (x86_64)
   2   │ Wed May 10 08:32:48 PM EEST 2023
   3   │ make -C /usr/lib/modules/6.4.0-rc1-1-mainline/build M=/var/lib/dkms/kvmfr/0.0.8/build modules
   4   │   CC [M]  /var/lib/dkms/kvmfr/0.0.8/build/kvmfr.o
   5   │ In file included from ./include/linux/linkage.h:7,
   6   │                  from ./arch/x86/include/asm/cache.h:5,
   7   │                  from ./include/linux/cache.h:6,
   8   │                  from ./arch/x86/include/asm/current.h:9,
   9   │                  from ./include/linux/sched.h:12,
  10   │                  from ./include/linux/ratelimit.h:6,
  11   │                  from ./include/linux/dev_printk.h:16,
  12   │                  from ./include/linux/device.h:15,
  13   │                  from /var/lib/dkms/kvmfr/0.0.8/build/kvmfr.c:24:
  14   │ /var/lib/dkms/kvmfr/0.0.8/build/kvmfr.c: In function ‘kvmfr_module_init’:
  15   │ ./include/linux/export.h:27:22: error: passing argument 1 of ‘class_create’ from incompatible pointer type [-Werror=incompatible-pointer-types]
  16   │    27 | #define THIS_MODULE (&__this_module)
  17   │       |                     ~^~~~~~~~~~~~~~~
  18   │       |                      |
  19   │       |                      struct module *
  20   │ /var/lib/dkms/kvmfr/0.0.8/build/kvmfr.c:542:32: note: in expansion of macro ‘THIS_MODULE’
  21   │   542 |   kvmfr->pClass = class_create(THIS_MODULE, KVMFR_DEV_NAME);
  22   │       |                                ^~~~~~~~~~~
  23   │ In file included from ./include/linux/device.h:31:
  24   │ ./include/linux/device/class.h:229:54: note: expected ‘const char *’ but argument is of type ‘struct module *’
  25   │   229 | struct class * __must_check class_create(const char *name);
  26   │       |                                          ~~~~~~~~~~~~^~~~
  27   │ /var/lib/dkms/kvmfr/0.0.8/build/kvmfr.c:542:19: error: too many arguments to function ‘class_create’
  28   │   542 |   kvmfr->pClass = class_create(THIS_MODULE, KVMFR_DEV_NAME);
  29   │       |                   ^~~~~~~~~~~~
  30   │ ./include/linux/device/class.h:229:29: note: declared here
  31   │   229 | struct class * __must_check class_create(const char *name);
  32   │       |                             ^~~~~~~~~~~~
  33   │ cc1: some warnings being treated as errors
  34   │ make[2]: *** [scripts/Makefile.build:252: /var/lib/dkms/kvmfr/0.0.8/build/kvmfr.o] Error 1
  35   │ make[1]: *** [Makefile:2026: /var/lib/dkms/kvmfr/0.0.8/build] Error 2
  36   │ make: *** [Makefile:7: all] Error 2
```

When I run make in the modules directory using your patch, I get the following error  

```
make -C /lib/modules/6.4.0-rc1-1-mainline/build M=/home/endre/Downloads/LookingGlass/module modules
  CC [M]  /home/endre/Downloads/LookingGlass/module/kvmfr.o
In file included from <command-line>:
././include/linux/kconfig.h:5:10: fatal error: generated/autoconf.h: No such file or directory
    5 | #include <generated/autoconf.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [scripts/Makefile.build:252: /home/endre/Downloads/LookingGlass/module/kvmfr.o] Error 1
make[1]: *** [Makefile:2026: /home/endre/Downloads/LookingGlass/module] Error 2
make: *** [Makefile:7: all] Error 2

```

Is there anything else that I need to do differently?

--- a/module/dkms.conf
+++ b/module/dkms.conf
@@ -1,5 +1,5 @@
 PACKAGE_NAME="kvmfr"
-PACKAGE_VERSION="0.0.8"
+PACKAGE_VERSION="0.0.9"
 BUILT_MODULE_NAME[0]="${PACKAGE_NAME}"
 MAKE[0]="make KDIR=${kernel_source_dir}"
 CLEAN="make KDIR=${kernel_source_dir} clean"

--- a/module/kvmfr.c
+++ b/module/kvmfr.c
@@ -569,7 +569,11 @@ static int __init kvmfr_module_init(void)
     goto out_free;
   }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 4, 0)
   kvmfr->pClass = class_create(THIS_MODULE, KVMFR_DEV_NAME);
+#else
+  kvmfr->pClass = class_create(KVMFR_DEV_NAME);
+#endif
   if (IS_ERR(kvmfr->pClass)) {
     printk(KERN_INFO "kvmfr: kvmfr_module_init: failed to create class!\n");
     goto out_unreg;

